### ISSUE TITLE
test(api-client): fix race ReferenceError: Cannot access 'sub' before init

### ISSUE THREAD

**Issue by petermetz** [hyperledger/cacti#2191]
**Describe the bug**

There might be a race condition in one of the tests that came about when the CI was running for the PR https://github.com/hyperledger/cactus/pull/2179 that is to fix the issue of https://github.com/hyperledger/cactus/issues/2158

The two things are MIGHT not be related at all and it's just happened so that the race condition was noticed randomly during the CI execution of this PR. Further investigation is needed to clarify this (unless the fix is easy enough without knowing the answer anyway).

```sh
2022-11-07T10:01:11.9443290Z   ● SocketIOApiClient Tests › Monitoring Tests › Sends stopMonitor and report error on monitor error
2022-11-07T10:01:11.9443607Z 
2022-11-07T10:01:11.9443814Z     thrown: "Exceeded timeout of 5000 ms for a test.
2022-11-07T10:01:11.9444495Z     Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."
2022-11-07T10:01:11.9444736Z 
2022-11-07T10:01:11.9444826Z       421 |     };
2022-11-07T10:01:11.9445026Z       422 |
2022-11-07T10:01:11.9445360Z     > 423 |     test("Sends stopMonitor and report error on monitor error", (done) => {
2022-11-07T10:01:11.9445648Z           |     ^
2022-11-07T10:01:11.9445915Z       424 |       expect.assertions(1);
2022-11-07T10:01:11.9446146Z       425 |
2022-11-07T10:01:11.9446423Z       426 |       // Connect client and server sockets
2022-11-07T10:01:11.9446589Z 
2022-11-07T10:01:11.9446999Z       at packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:423:5
2022-11-07T10:01:11.9447619Z       at packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:416:3
2022-11-07T10:01:11.9448455Z       at Object.<anonymous> (packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:194:1)
2022-11-07T10:01:11.9448735Z 
2022-11-07T10:01:11.9449078Z   ● SocketIOApiClient Tests › Monitoring Tests › Sends stopMonitor on unsubscribe
2022-11-07T10:01:11.9449303Z 
2022-11-07T10:01:11.9449474Z     thrown: "Exceeded timeout of 5000 ms for a test.
2022-11-07T10:01:11.9450008Z     Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."
2022-11-07T10:01:11.9450244Z 
2022-11-07T10:01:11.9450338Z       448 |     });
2022-11-07T10:01:11.9450536Z       449 |
2022-11-07T10:01:11.9450846Z     > 450 |     test("Sends stopMonitor on unsubscribe", (done) => {
2022-11-07T10:01:11.9451097Z           |     ^
2022-11-07T10:01:11.9451379Z       451 |       // Connect client and server sockets
2022-11-07T10:01:11.9451715Z       452 |       testServer.on("connection", (socket) => {
2022-11-07T10:01:11.9452082Z       453 |         log.debug("Server socket connected", socket.id);
2022-11-07T10:01:11.9452270Z 
2022-11-07T10:01:11.9452659Z       at packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:450:5
2022-11-07T10:01:11.9453277Z       at packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:416:3
2022-11-07T10:01:11.9453952Z       at Object.<anonymous> (packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:194:1)
2022-11-07T10:01:11.9454228Z 
2022-11-07T10:01:11.9454648Z   ● SocketIOApiClient Tests › Monitoring Tests › Repeat watchBlocksV1 after previous monitor observable finished
2022-11-07T10:01:11.9455031Z 
2022-11-07T10:01:11.9455309Z     ReferenceError: Cannot access 'sub' before initialization
2022-11-07T10:01:11.9455499Z 
2022-11-07T10:01:11.9455617Z       692 |               } else {
2022-11-07T10:01:11.9455944Z       693 |                 log.info("2nd observer received event:", ev);
2022-11-07T10:01:11.9456262Z     > 694 |                 sub.unsubscribe();
2022-11-07T10:01:11.9456503Z           |                 ^
2022-11-07T10:01:11.9456754Z       695 |                 resolve(ev);
2022-11-07T10:01:11.9456967Z       696 |               }
2022-11-07T10:01:11.9457180Z       697 |             },
2022-11-07T10:01:11.9457310Z 
2022-11-07T10:01:11.9457740Z       at Object.next (packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:694:17)
2022-11-07T10:01:11.9458410Z       at Object.next (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/Subscriber.js:161:21)
2022-11-07T10:01:11.9459180Z       at SafeSubscriber.Object.<anonymous>.Subscriber._next (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/Subscriber.js:101:26)
2022-11-07T10:01:11.9459983Z       at SafeSubscriber.Object.<anonymous>.Subscriber.next (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/Subscriber.js:72:18)
2022-11-07T10:01:11.9460894Z       at ReplaySubject.Object.<anonymous>.ReplaySubject._subscribe (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/ReplaySubject.js:54:24)
2022-11-07T10:01:11.9461730Z       at ReplaySubject.Object.<anonymous>.Observable._trySubscribe (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/Observable.js:41:25)
2022-11-07T10:01:11.9462543Z       at ReplaySubject.Object.<anonymous>.Subject._trySubscribe (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/Subject.js:120:47)
2022-11-07T10:01:11.9463201Z       at packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/Observable.js:35:31
2022-11-07T10:01:11.9463873Z       at Object.errorContext (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/util/errorContext.js:22:9)
2022-11-07T10:01:11.9464667Z       at ReplaySubject.Object.<anonymous>.Observable.subscribe (packages/cactus-api-client/node_modules/rxjs/dist/cjs/internal/Observable.js:26:24)
2022-11-07T10:01:11.9465410Z       at secondResults (packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:687:50)
2022-11-07T10:01:11.9466114Z       at Object.<anonymous> (packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:685:35)
2022-11-07T10:01:11.9466423Z 
2022-11-07T10:01:11.9466847Z   ● SocketIOApiClient Tests › Monitoring Tests › Repeat watchBlocksV1 after previous monitor observable finished
2022-11-07T10:01:11.9467109Z 
2022-11-07T10:01:11.9467281Z     thrown: "Exceeded timeout of 5000 ms for a test.
2022-11-07T10:01:11.9467909Z     Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."
2022-11-07T10:01:11.9468151Z 
2022-11-07T10:01:11.9468257Z       641 |     });
2022-11-07T10:01:11.9468457Z       642 |
2022-11-07T10:01:11.9468836Z     > 643 |     test("Repeat watchBlocksV1 after previous monitor observable finished", async () => {
2022-11-07T10:01:11.9469144Z           |     ^
2022-11-07T10:01:11.9469431Z       644 |       // Server side logic for sending events
2022-11-07T10:01:11.9469770Z       645 |       testServer.on("connection", (socket) => {
2022-11-07T10:01:11.9470122Z       646 |         log.debug("Server socket connected", socket.id);
2022-11-07T10:01:11.9470312Z 
2022-11-07T10:01:11.9470697Z       at packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:643:5
2022-11-07T10:01:11.9471315Z       at packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:416:3
2022-11-07T10:01:11.9471997Z       at Object.<anonymous> (packages/cactus-api-client/src/test/typescript/unit/socketio-api-client.test.ts:194:1)
2022-11-07T10:01:11.9472272Z 
```

**To Reproduce**

Unclear. Large number of executions will probably trigger it happening eventually.

**Expected behavior**

The test case passes or fails consistently.

**Logs/Stack traces**

Full log:
[2022-11-07-ci-cactus-api-client-crash.log](https://github.com/hyperledger/cactus/files/9954870/2022-11-07-ci-cactus-api-client-crash.log)

**Cloud provider or hardware configuration:**

GitHub Action Runner

**Operating system name, version, build:**

Ubuntu-20.04

**Hyperledger Cactus release version or commit (git rev-parse --short HEAD):**

`v1.1.0`

**Hyperledger Cactus Plugins/Connectors Used**

N/A

**Additional context**

cc: @outSH @aldousalvarez

**Comment by aldousalvarez** [comment]
@petermetz  Apologies, while doing the fix on upgrade migration of typescript to nodeNext the path of the imports changed. Based on the suggested fix observers must be used thus the issue. Upon checking, the observed method returns the observers.length and I've already implemented it to <**packages/cactus-api-client/src/main/typescript/socketio-api-client.ts**> which is now passing see the latest [commit](https://github.com/hyperledger/cactus/pull/2179/commits/7f6493e6a866b98a51b72664b5658e7fa867a42d).

**Comment by outSH** [comment]
I think indeed this has been fixed in #2179, have not seen any api-client failures since then - closing
